src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -426,7 +429,15 @@ static CHIP_ERROR GetAddrInfo(void * context, DnssdResolveCallback callback, uin
         err = TXTRecordGetItemAtIndex(txtLen, txtRecord, i, kDnssdKeyMaxSize, key, &valueLen, &valuePtr);
         VerifyOrReturnError(CheckForSuccess(sdCtx, __func__, err, true), CHIP_ERROR_INTERNAL);
 
-        strncpy(value, reinterpret_cast<const char *>(valuePtr), valueLen);
+        if (valueLen >= sizeof(value))
+        {
+            // Truncation, but nothing better we can do
+            valueLen = sizeof(value) - 1;
+        }
+        if (valueLen > 1)
+        {
+            memcpy(value, valuePtr, valueLen - 1);
+        }
```

Comment:
I don't get this part.  Shouldn't this just be:
```suggestion
          memcpy(value, valuePtr, valueLen);
```
?  We already made sure that `valueLen` leaves 1 byte of space in `value`; why would we only copy `valueLen - 1` bytes?